build-locally.sh

```diff
@@ -0,0 +1,251 @@
+    info "Over-ridden by caller using the LOGS_DIR variable."
+fi
+
+info "Using source code at ${source_dir}"
+cd ${BASEDIR}/${source_dir}
+if [[ "${DEBUG}" == "1" ]]; then
+    OPTIONAL_DEBUG_FLAG="-debug"
+else
+    OPTIONAL_DEBUG_FLAG="-info"
+fi
+
+
+log_file=${LOGS_DIR}/${project}.txt
+info "Log will be placed at ${log_file}"
+date > ${log_file}
+
+h2 "Checking dependencies are present..."
+declare -a required_files=(
+${WORKSPACE_DIR}/${project}/dev.galasa.uber.obr/pom.template
+${WORKSPACE_DIR}/framework/release.yaml 
+${WORKSPACE_DIR}/extensions/release.yaml 
+${WORKSPACE_DIR}/managers/release.yaml 
+${WORKSPACE_DIR}/obr/release.yaml 
+)
+for required_file in "${required_files[@]}"
+do
+    if [[ -e "${required_file}" ]]; then
+        success "OK - File ${required_file} is present."
+    else 
+        error "File ${required_file} is required, but missing. Clone the sibling project to make sure it exists."
+    fi
+done
+
+h2 "Generating a pom.xml from a template, using all the versions of everything..."
+
+cat << EOF >> ${log_file}
+Using command:
+
+${GALASA_BUILD_TOOL_PATH} template \
+'--releaseMetadata' ${WORKSPACE_DIR}/framework/release.yaml \
+'--releaseMetadata' ${WORKSPACE_DIR}/extensions/release.yaml \
+'--releaseMetadata' ${WORKSPACE_DIR}/managers/release.yaml \
+'--releaseMetadata' ${WORKSPACE_DIR}/obr/release.yaml \
+'--template' pom.template \
+'--output' pom.xml \
+'--obr' \
+2>&1 >> ${log_file}
+
+EOF
+
+cd ${WORKSPACE_DIR}/${project}/dev.galasa.uber.obr
+
+# What's the architecture-variable name of the build tool we want for this local build ?
+export ARCHITECTURE=$(uname -m) # arm64 or amd64
+export GALASA_BUILD_TOOL_NAME=galasabld-darwin-${ARCHITECTURE}
+
+# Favour the galasabld tool if it's on the path, else use a locally-built version or fail if not available.
+GALASABLD_ON_PATH=$(which galasabld)
+rc=$?
+if [[ "${rc}" == "0" ]]; then
+    info "Using the 'galasabld' tool which is on the PATH"
+    GALASA_BUILD_TOOL_PATH=${GALASABLD_ON_PATH}
+else
+    GALASABLD_ON_PATH=$(which $GALASA_BUILD_TOOL_NAME)
+    rc=$?
+    if [[ "${rc}" == "0" ]]; then
+        info "Using the '$GALASA_BUILD_TOOL_NAME' tool which is on the PATH"
+        GALASA_BUILD_TOOL_PATH=${GALASABLD_ON_PATH}
+    else
+        info "The galasa build tool 'galasabld' or '$GALASA_BUILD_TOOL_NAME' is not on the path."
+        export GALASA_BUILD_TOOL_PATH=${WORKSPACE_DIR}/buildutils/bin/${GALASA_BUILD_TOOL_NAME}
+        if [[ ! -e ${GALASA_BUILD_TOOL_PATH} ]]; then
+            error "Cannot find the $GALASA_BUILD_TOOL_NAME tools on locally built workspace."
+            info "Try re-building the buildutils project"
+        else
+            info "Using the $GALASA_BUILD_TOOL_NAME tool at ${GALASA_BUILD_TOOL_PATH}"
+        fi
+    fi
+fi
+
+# Check local build version
+export GALASA_BUILD_TOOL_PATH=${WORKSPACE_DIR}/buildutils/bin/${GALASA_BUILD_TOOL_NAME}
+info "Using galasabld tool ${GALASA_BUILD_TOOL_PATH}"
```

Comment:
Did we want a message to appear if cannot find on path or built? Just to let user know it cant find a required tool

Reply:
already have:
```
info "The galasa build tool 'galasabld' or '$GALASA_BUILD_TOOL_NAME' is not on the path."
```

Reply:
I am blind, approving